eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -0,0 +1,247 @@
+
+            return;
+        }
+
+        $output->writeln(sprintf(
+            '<info>Found %d Content IDs matching given criteria.</info>',
+            $contentIdsCount
+        ));
+
+        $removedVersionsCounter = 0;
+
+        foreach ($contentIds as $contentId) {
+            try {
+                $contentInfo = $this->contentService->loadContentInfo((int) $contentId);
+                $versions = $this->contentService->loadVersions($contentInfo);
+                $versionsCount = count($versions);
+
+                $output->writeln(sprintf(
+                    '<info>Content %d has %d version(s)</info>',
+                    (int) $contentId,
+                    $versionsCount
+                ), Output::VERBOSITY_VERBOSE);
+
+                $versions = array_slice(array_reverse($versions), $keep);
+
+                $removeAll = $status === self::VERSION_ALL;
+                $removeDrafts = $status === self::VERSION_DRAFT;
+                $removeArchived = $status === self::VERSION_ARCHIVED;
+
+                /** @var \eZ\Publish\API\Repository\Values\Content\VersionInfo $version */
+                foreach ($versions as $version) {
+                    if (
+                        ($removeAll && $version->status !== VersionInfo::STATUS_PUBLISHED) ||
+                        ($removeDrafts && $version->status === VersionInfo::STATUS_DRAFT) ||
+                        ($removeArchived && $version->status === VersionInfo::STATUS_ARCHIVED)
+                    ) {
+                        $this->contentService->deleteVersion($version);
+                        ++$removedVersionsCounter;
+                        $output->writeln(sprintf(
+                            "Content's (%d) version (%d) has been deleted.",
+                            $contentInfo->id,
+                            $version->id
+                        ), Output::VERBOSITY_VERBOSE);
+                    }
+                }
+            } catch (Exception $e) {
+                $output->writeln(sprintf(
+                    '<error>%s</error>',
+                    $e->getMessage()
+                ));
+            }
+        }
+
+        $output->writeln(sprintf(
+            '<info>Removed %d unwanted contents version(s).</info>',
+            $removedVersionsCounter
+        ));
+    }
+
+    /**
+     * @param int $keep
+     * @param string $status
+     *
+     * @return array
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     */
+    protected function getObjectsIds($keep, $status)
+    {
+        $query = $this->connection->createQueryBuilder()
+                ->select('c.id')
+                ->from('ezcontentobject', 'c')
+                ->join('c', 'ezcontentobject_version', 'v', 'v.contentobject_id = c.id')
+                ->groupBy('c.id', 'v.status')
+                ->having('count(c.id) > :keep');
+        $query->setParameter('keep', $keep);
```

Comment:
Tip for the future: You are able to create and bind parameters at once by using `\Doctrine\DBAL\Query\QueryBuilder::createNamedParameter`

Reply:
True, but then you need to use expr(), which sometimes make it harder to read the query :P

Reply:
Since this tip is just for the future, I decided to keep the code as it is. Maybe we will like to improve it at some point later.

Reply:
So ready for QA then?